Detectors/HMPID/calibration/src/HMPIDDCSProcessor.cxx

```diff
@@ -0,0 +1,759 @@
+    std::vector<TF1>::iterator itTout = arNmean.begin() + 6 * iCh + 2 * iRad + 1;
+    arNmean.insert(itTout, *(pTout.get()));
+  } else {
+    LOG(info) << Form("No entries in temp-out for ch%irad%i", iCh, iRad);
+  }
+}
+
+// returns false if the element in array of DPCOM-vector has no entries
+void HMPIDDCSProcessor::finalizeTempIn(int iCh, int iRad)
+{
+
+  if (dpVecTempIn[3 * iCh + iRad].size() != 0) {
+    cntTin = 0;
+
+    auto minTime = getMinTime(dpVecTempIn[3 * iCh + iRad]);
+    auto maxTime = getMaxTime(dpVecTempIn[3 * iCh + iRad]);
+
+    std::unique_ptr<TGraph> pGrTIn;
+    pGrTIn.reset(new TGraph);
+
+    for (DPCOM dp : dpVecTempIn[3 * iCh + iRad]) {
+      auto dpVal = o2::dcs::getValue<double>(dp);
+      auto time = dp.data.get_epoch_time(); //-minTime
+      pGrTIn->SetPoint(cntTin++, time, dpVal);
+    }
+    // maxTime -= minTime;
+    // minTime = 0;
+    std::unique_ptr<TF1> pTin;
+    pTin.reset(new TF1(Form("Tin%i%i", iCh, iRad), "[0]+[1]*x", minTime, maxTime));
+
+    if (cntTin == 1) {
+      pGrTIn->GetPoint(0, xP, yP);
+      pTin->SetParameter(0, yP);
+      pTin->SetParameter(1, 0);
+    } else {
+      pGrTIn->Fit(Form("Tin%i%i", iCh, iRad), "Q");
+    }
+
+    pTin->SetTitle(Form("Temp-In Fit Chamber%i Radiator%i; Time [ms]; Temp [C]", iCh, iRad));
+    std::vector<TF1>::iterator itTin = arNmean.begin() + 6 * iCh + 2 * iRad;
+    arNmean.insert(itTin, *(pTin.get()));
+
+  } else {
+    LOG(info) << Form("No entries in temp-in for ch%irad%i", iCh, iRad);
+  }
+}
+
+// returns false if the element in array of DPCOM-vector has no entries
+std::unique_ptr<TF1> HMPIDDCSProcessor::finalizeHv(int iCh, int iSec)
+{
+  std::unique_ptr<TF1> pHvTF;
+  if (dpVecHV[3 * iCh + iSec].size() != 0) {
+
+    std::unique_ptr<TGraph> pGrHV;
+    pGrHV.reset(new TGraph);
+    cntHV = 0;
+
+    hvFirstTime = getMinTime(dpVecHV[3 * iCh + iSec]);
+    hvLastTime = getMaxTime(dpVecHV[3 * iCh + iSec]);
+
+    for (DPCOM dp : dpVecHV[3 * iCh + iSec]) {
+      auto dpVal = o2::dcs::getValue<double>(dp);
+      auto time = dp.data.get_epoch_time(); //- hvFirstTime
+      pGrHV->SetPoint(cntHV++, time, dpVal);
+    }
+    // hvLastTime -= hvFirstTime;
+    // hvFirstTime = 0;
+    if (cntHV == 1) {
+      pGrHV->GetPoint(0, xP, yP);
+      (pHvTF).reset(new TF1(Form("HV%i_%i", iCh, iSec), Form("%f", yP), hvFirstTime, hvLastTime));
+      pArrHv[3 * iCh + iSec] = *(pHvTF.get());
+    } else {
+      (pHvTF).reset(new TF1(Form("HV%i_%i", iCh, iSec), "[0]+x*[1]", hvFirstTime, hvLastTime));
+      pGrHV->Fit(Form("HV%i_%i", iCh, iSec), "Q");
+      pArrHv[3 * iCh + iSec] = *(pHvTF.get());
+    }
+
+    return pHvTF;
+  }
+  LOG(info) << Form("No entries in HV for ch%isec%i", iCh, iSec);
+  return pHvTF;
+}
+
+// Process all arrays of DPCOM-vectors, perform fits
+// Called At the end of the Stream of DPs, in
+// HMPIDDCSDataProcessorSpec::endOfStream() function
+void HMPIDDCSProcessor::finalize()
+{
```

Comment:
As far as I can see, you create a CCDB object only in the endOfStream, but this means never, as the workflow is running permanently, w/o any correlation with data taking (SOR/EOR).
In case you need to perform some action at the EOR, you can use https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/DCS/include/DetectorsDCS/RunStatusChecker.h, see how EMCAL does this: https://github.com/AliceO2Group/AliceO2/blob/3d5a983fbadf879127fbde5d17e6a85b350c985d/Detectors/EMCAL/workflow/include/EMCALWorkflow/EMCALDCSDataProcessorSpec.h#L130-L146